jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -171,11 +171,12 @@ protected void drawFill(Graphics2D g) {
 
   @Override
   public boolean contains(Point2D.Double p, double scaleDenominator) {
+    boolean result = false;
     double tolerance =
         Math.max(1f, 2 * AttributeKeys.getPerpendicularHitGrowth(this, scaleDenominator));
     if (isClosed() || attr().get(FILL_COLOR) != null && attr().get(UNCLOSED_PATH_FILLED)) {
```

Comment:
The method currently has 10 conditionals, which is a lot. Another refactoring could be done to reduce that complexity. Extract Method could be used twice here inside the main "if... else if" statement. Two private methods could be made for what is inside that statement. Each private method could return a boolean value. The value would then go in the "result" variable.